### PR TITLE
NUTCH-2374 Upgrade Nutch 2.X to Gora 0.7

### PR DESCRIPTION
**Pull request by cloudysunny14**

### CODE REVIEW

**Comment by cloudysunny14** [comment]
Hi @lewismc. Thank you for your review.

Sorry for my late reply.
I have tried the new stores of Gora 0.7, results are below.

| DataStore | Result | Description |
|-----------|---------|-------------|
| JCache     | NG         |  Preparing patch for Gora |
| Infinispan  | NG        | Under investigation. Error occurred in client |
| CouchDB  | NG         | Problem with update the document. Is MapReduce unsupported?※ |
| DynamoDB | NG       | Conflicting with AWS client library. Is MapReduce unsupported?※ |

※There is no MapReduce integration test in the repository of Gora 0.7.

JCacheStore works by rebuild classes of storage and applying patch to Gora to disable CacheEntryListener,
Others are under investigation.

By the way, with HBaseStore at this PR code base, It worked.
InfinispanStore is investigating, but it takes much time. May I assume that CouchDB and DynamoDB are out of the scope Nutch 2.4?

**Comment by lewismc** [comment]
@cloudysunny14 what does RESULT NG mean?

> By the way, with HBaseStore at this PR code base, It worked.

great

> InfinispanStore is investigating, but it takes much time. May I assume that CouchDB and DynamoDB are out of the scope Nutch 2.4?

Yes that is absolutely fine.

> JCacheStore works by rebuild classes of storage and applying patch to Gora to disable CacheEntryListener,

Can you point me to this PR please?

File: src/java/org/apache/nutch/storage/Host.java [hunk 1]
```diff
@@ -1,40 +1,23 @@
...
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- ******************************************************************************/
 /**
  * Autogenerated by Avro
- * 
+ *
```
Comment (lewismc) [inline src/java/org/apache/nutch/storage/Host.java]: Can you also please make sure that when these classes are recompiled that a new license header is manually added to each file? Thank you

**Comment by lewismc** [comment]
@cloudysunny14 ping, I would like to work with you to shepherd this code into 2.X branch.

**Comment by cloudysunny14** [comment]
Hi @lewismc.
I appreciate your reply, and I'm so sorry for not replying sooner.

> what does RESULT NG mean?

OK: The crawl cycle finishes normally.
NG: The crawl cycle interruption due to an error.

> Can you point me to this PR please?

I am planning to make CacheEntryListener disable by configuration but it is under consideration.
I will create issue about JCacheStore in the Gora's JIRA.

Also, added license header according to your feedback.

Thank you.

**Comment by lewismc** [comment]
OK I will also test some of the datastores shortly, thank you

**Comment by lewismc** [comment]
testing is continuous now... will provide more updates @cloudysunny14 thank you

**Comment by tulay** [comment]
Why is this still open? Gora 0.8 is out. I am using it to work against mongodb. How do you test against this for all supported storage btw?

**Comment by lewismc** [comment]
> How do you test against for all supported storage btw?

 @tulay we rely upon community input and testing between Gora + Nutch communities.

**Comment by tulay** [comment]
@lewismc then what do you say if I update to 0.8 and wait for getting feedback?

**Comment by lewismc** [comment]
+1 that would be great. We could push a Nutch 2.4 release candidate after that.

**Pull request merged**